### PR TITLE
build_style/gem: add

### PR DESCRIPTION
**Pull request by maxice8**
This adds a build_style for dealing with gemspec files and has conveniences for rubygems

Please do not merge this as there is still much to be decided

### CODE REVIEW

File: Manual.md [hunk 1]
```diff
@@ -781,6 +781,17 @@ with the character `r` in the `version` variable. The `distfiles`
 location will automatically be set as well as the package made to depend
 on `R`.
 
+- `gem` For packages that are installed using gemspec files or gems
+from rubygems. The build style automatically sets `distfiles` to the
```
Comment (Cogitri) [inline Manual.md]: IMO "The build style" is superfluous

File: Manual.md [hunk 2]
```diff
@@ -781,6 +781,17 @@ with the character `r` in the `version` variable. The `distfiles`
...
+rubygems name of the package if not set manually. The `gem_cmd`
```
Comment (Cogitri) [inline Manual.md]: Uhh...the "rubygems" name is exactly what? The pkgname, but without `ruby-`?

Comment (Cogitri) [reply Manual.md]: Also, IMHO `gem_bin` might be a little clearer, but either is fine.

File: Manual.md [hunk 3]
```diff
@@ -781,6 +781,17 @@ with the character `r` in the `version` variable. The `distfiles`
...
+variable can be used to set the name of the gem binary.
+
+The build style requires the `pkgname` variable to be their name on
+RubyGems with ruby- prefixed. The `distfiles` location is automatically
```
Comment (Cogitri) [inline Manual.md]: Maybe add a link to RubyGems here?

File: Manual.md [hunk 4]
```diff
@@ -781,6 +781,17 @@ with the character `r` in the `version` variable. The `distfiles`
...
+set, ruby is added to `hostmakedepends` and `depends`, and the build
+process is extensively modified to not use a `wrksrc` but instead call
+`gem` into the downloaded gem file directly.
```
Comment (Cogitri) [inline Manual.md]: Hm, how would one go about doing custom phases then? Would one still go into ${wrksrc} e.g. to do sed calls?

File: common/build-style/gem.sh [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+RbConfig::MAKEFILE_CONFIG['LD'] = ENV['LD'] if ENV['LD']
+RbConfig::MAKEFILE_CONFIG['CFLAGS'] = ENV['CFLAGS'] if ENV['CFLAGS']
+RbConfig::MAKEFILE_CONFIG['CXXFLAGS'] = ENV['CXXFLAGS'] if ENV['CXXFLAGS']
+EOF
+
+		# Patch all instances of extconf that use create_makefile
+		for f in $(find . -type f -name 'extconf.rb'); do
+			if [ ! -f ${f}.orig ]; then
+				# Ignore extconf files that do not create makefiles
+				grep create_makefile $f || continue
```
Comment (Cogitri) [inline common/build-style/gem.sh]: Maybe use `grep -q`

File: common/build-style/gem.sh [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+			i686) _HOST_PLATFORM=i686-linux ;;
+			i686-musl) _HOST_PLATFORM=i686-linux-musl ;;
+			armv7l) _HOST_PLATFORM=/armv7l-linux-eabihf ;;
+			armv6l) _HOST_PLATFORM=arm-linux-eabihf ;;
+			armv7l-musl) _HOST_PLATFORM=armv7l-linux-musleabihf ;;
+			armv6l-musl) _HOST_PLATFORM=arm-linux-musleabihf ;;
+			aarch64) _HOST_PLATFORM=aarch64-linux ;;
+			aarch64-musl) _HOST_PLATFORM=aarch64-linux-musl ;;
+		esac
+	fi
```
Comment (Cogitri) [inline common/build-style/gem.sh]: Hm, shouldn't this be in `common/environment/build-style`?

File: common/build-style/gem.sh [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+
+	if [ "$CROSS_BUILD" ]; then
+		# Path to the extensions on a package, ruby installs against the platform
+		# of the host, so we have to move them to the correct place
+		_HOST_EXT_DIR="${DESTDIR}/${_GEMDIR}/extensions/${_HOST_PLATFORM}"
+		_TARGET_EXT_DIR="${DESTDIR}/${_GEMDIR}/extensions/${_TARGET_PLATFORM}"
+
+		if [ -d ${_HOST_EXT_DIR} -a "${_HOST_EXT_DIR}" != "${_TARGET_EXT_DIR}" ]; then
+			mv ${_HOST_EXT_DIR} ${_TARGET_EXT_DIR}
+		fi
```
Comment (Cogitri) [inline common/build-style/gem.sh]: When would `_HOST_EXT_DIR` be `_TARGET_EXT_DIR` during cross compiling?

File: Manual.md [hunk 8]
```diff
@@ -781,6 +781,12 @@ with the character `r` in the `version` variable. The `distfiles`
 location will automatically be set as well as the package made to depend
 on `R`.
 
+- `gem` For packages that are installed using gemspec files or gems
+from rubygems. It is expected that packages that use this build style
+follow the ruby-\<pkgname\> naming convention with pkgname being the
+name of the package according to rubygems. The `gem_cmd` variable can
```
Comment (Cogitri) [inline Manual.md]: Please add a link to RubyGems here.

File: common/build-style/gem.sh (hunk 7 above)
Comment (maxice8) [reply common/build-style/gem.sh]: When ?

File: common/build-style/gem.sh (hunk 6 above)
Comment (maxice8) [reply common/build-style/gem.sh]: No, it is only used during build

File: common/build-style/gem.sh (hunk 5 above)
Comment (maxice8) [reply common/build-style/gem.sh]: done

File: common/build-style/gem.sh (hunk 6 above)
Comment (Cogitri) [reply common/build-style/gem.sh]: Ah, okay.

File: common/build-style/gem.sh [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+		for f in $(find . -type f -name 'extconf.rb'); do
+			if [ ! -f ${f}.orig ]; then
+				# Ignore extconf files that do not create makefiles
+				grep -q create_makefile $f || continue
+				# Create a backup which we will restore later
+				cp $f ${f}.orig
+				# Patch extconf.rb for cross compile
+				cat append ${f}.orig >> $f
+			fi
+		done
```
Comment (Cogitri) [inline common/build-style/gem.sh]: Hm, maybe doing `patch` and later on `patch -R` would be a little more elegant?

File: common/build-style/gem.sh (hunk 7 above)
Comment (Cogitri) [reply common/build-style/gem.sh]: How would these two ever match during cross.

File: Manual.md [hunk 10]
```diff
@@ -781,6 +781,12 @@ with the character `r` in the `version` variable. The `distfiles`
 location will automatically be set as well as the package made to depend
 on `R`.
 
+- `gem` For packages that are installed using gemspec files or gems
+from [RubyGems](https://rubygems.org/). It is expected that packages
```
Comment (Vaelatern) [inline Manual.md]: `Packages that use this build style must follow the ruby-<pkgname> naming convention, with....`

Comment (maxice8) [reply Manual.md]: fixed, thanks

**Comment by Vaelatern** [comment]
I am unfamiliar enough with gem to be uncomfortable green-lighting this without my extensive review, but I don't wish to block the merge.

**Comment by maxice8** [comment]
On Sat, Oct 27, 2018 at 08:29:33PM -0700, Toyam Cox wrote:
> I am unfamiliar enough with gem to be uncomfortable green-lighting this without my extensive review, but I don't wish to block the merge.
> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/4193#issuecomment-433673319

Take all the time you want, There are still some things that need to be decided before this is done

File: common/build-style/gem.sh (hunk 9 above)
Comment (Cogitri) [reply common/build-style/gem.sh]: Just write the stuff you write into append into a patch file, do `patch -i $patch_file` when you want to modify it and `patch -R -i $patch_file` when you want to revert it.

File: common/build-style/gemspec.sh [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+				# Patch extconf.rb for cross compile
+				cat append ${f}.orig >> $f
+			fi
+		done
+	fi
+
+	$gem_cmd build ${pkgname#ruby-}.gemspec
+
+	if [ "$CROSS_BUILD" ]; then
+		# Restrore previous extconf.rb which we ship.
```
Comment (Cogitri) [inline common/build-style/gemspec.sh]: typo

File: common/build-style/gemspec.sh [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+	: ${gem_cmd:=gem}
+
+	# Some packages keep Gem.gemspec as the name instead of the proper
+	# $pkgname.gemspec, fix those
+	if [ -f Gem.gemspec ]; then
+		mv Gem.gemspec ${pkgname#ruby-}.gemspec
+	fi
+
+	# Replace name and version just in case they try something funny like
+	# requring RELEASE to be on the environment to not append -rc0 to version
```
Comment (Cogitri) [inline common/build-style/gemspec.sh]: typo

File: common/build-style/gemspec.sh [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+	_GEMDIR="$($gem_cmd env gemdir)"
+
+	if [ "$CROSS_BUILD" ]; then
+		# Convert to the platforms used in our ruby packages
+		case "$XBPS_TARGET_MACHINE" in
+			x86_64) _TARGET_PLATFORM=x86_64-linux ;;
+			x86_64-musl) _TARGET_PLATFORM=x86_64-linux-musl ;;
+			i686) _TARGET_PLATFORM=i686-linux ;;
+			i686-musl) _TARGET_PLATFORM=i686-linux-musl ;;
+			armv7l) _TARGET_PLATFORM=/armv7l-linux-eabihf ;;
```
Comment (Cogitri) [inline common/build-style/gemspec.sh]: Is that `/` supposed to be here?

File: common/build-style/gemspec.sh [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+			aarch64) _TARGET_PLATFORM=aarch64-linux ;;
+			aarch64-musl) _TARGET_PLATFORM=aarch64-linux-musl ;;
+		esac
+
+		case "$XBPS_MACHINE" in
+			x86_64) _HOST_PLATFORM=x86_64-linux ;;
+			x86_64-musl) _HOST_PLATFORM=x86_64-linux-musl ;;
+			i686) _HOST_PLATFORM=i686-linux ;;
+			i686-musl) _HOST_PLATFORM=i686-linux-musl ;;
+			armv7l) _HOST_PLATFORM=/armv7l-linux-eabihf ;;
```
Comment (Cogitri) [inline common/build-style/gemspec.sh]: Same here

File: common/environment/build-style/gem.sh [hunk 15]
```diff
@@ -0,0 +1,5 @@
+lib32disabled=yes
+hostmakedepends+=" ruby"
+depends+= "ruby"
```
Comment (Cogitri) [inline common/environment/build-style/gem.sh]: I think the space is at the wrong place here

File: Manual.md [hunk 16]
```diff
@@ -781,6 +781,18 @@ with the character `r` in the `version` variable. The `distfiles`
 location will automatically be set as well as the package made to depend
 on `R`.
 
+- `gemspec` For packages that are built into gems using .gemspec files
+and then packaged. This build style is meant for packages that have
+C extensions and as such require patching to support cross compilation.
```
Comment (Cogitri) [inline Manual.md]: `and as such require patching to support cross compilation.`

Is it really necessary to include this here? Sounds as if this is something the user of this build_style has to worry about the way it's currently worded.

File: Manual.md [hunk 17]
```diff
@@ -781,6 +781,18 @@ with the character `r` in the `version` variable. The `distfiles`
...
+
+- `gem` For packages that are installed using gems from [RubyGems](https://rubygems.org/)
+or other providers. Packages that use this build style must follow the ruby-`pkgname`
+naming convention with `pkgname` being the name of the package according
+build style. The `gem_cmd` variable can be used to set the name of the gem binary.
```
Comment (Cogitri) [inline Manual.md]: `... being the name of the package according build style.`

Eh?

File: Manual.md [hunk 18]
```diff
@@ -781,6 +781,18 @@ with the character `r` in the `version` variable. The `distfiles`
...
+
+Packages under this build style are always `noarch`, if your gem provides compiled
```
Comment (Cogitri) [inline Manual.md]: `Packages using this...` sounds better IMO

Comment (Cogitri) [reply Manual.md]: Also, s/compiled/compiles/

File: Manual.md [hunk 19]
```diff
@@ -781,6 +781,18 @@ with the character `r` in the `version` variable. The `distfiles`
...
+extensions use the `gemspec` build style that can handle that.
```
Comment (Cogitri) [inline Manual.md]: _C_ extensions?

Comment (Cogitri) [reply Manual.md]: `that can handle that.`

Somehow that sounds pretty "meh" to me. Maybe just say "use the `gemspec` build style instead".

File: common/build-style/gem.sh [hunk 20]
```diff
@@ -0,0 +1,23 @@
...
+	local _GEMDIR
+
+	_GEMDIR="$($gem_cmd env gemdir)"
+
+	$gem_cmd install \
+		--local \
+		--install-dir ${DESTDIR}/${_GEMDIR} \
+		--bindir ${DESTDIR}/usr/bin \
+		--ignore-dependencies \
+		--verbose \
```
Comment (Cogitri) [inline common/build-style/gem.sh]: Although I'm all for being verbose I think void usually doesn't use the verbose mode.

File: common/build-style/gemspec.sh [hunk 21]
```diff
@@ -0,0 +1,131 @@
...
+			x86_64-musl) _TARGET_PLATFORM=x86_64-linux-musl ;;
+			i686) _TARGET_PLATFORM=i686-linux ;;
+			i686-musl) _TARGET_PLATFORM=i686-linux-musl ;;
+			armv7l) _TARGET_PLATFORM=armv7l-linux-eabihf ;;
+			armv6l) _TARGET_PLATFORM=arm-linux-eabihf ;;
+			armv7l-musl) _TARGET_PLATFORM=armv7l-linux-musleabihf ;;
+			armv6l-musl) _TARGET_PLATFORM=arm-linux-musleabihf ;;
+			aarch64) _TARGET_PLATFORM=aarch64-linux ;;
+			aarch64-musl) _TARGET_PLATFORM=aarch64-linux-musl ;;
+		esac
```
Comment (Cogitri) [inline common/build-style/gemspec.sh]: What about armv{6,7}hf? I can't keep up with all of those arm variants.

File: common/environment/build-style/gem.sh [hunk 22]
```diff
@@ -0,0 +1,10 @@
+lib32disabled=yes
+hostmakedepends+=" ruby"
+depends+=" ruby"
+skip_extraction="${pkgname}-${version}.gem"
+noarch=yes
+
+# deafult to rubygems
```
Comment (Cogitri) [inline common/environment/build-style/gem.sh]: typo

File: Manual.md (hunk 18 above)
Comment (maxice8) [reply Manual.md]: provides compiles extensions ?

File: common/build-style/gem.sh (hunk 20 above)
Comment (maxice8) [reply common/build-style/gem.sh]:
```
$ gem install --install-dir /tmp/asciidoctor asciidoctor-1.5.4.gem
Successfully installed asciidoctor-1.5.4
1 gem installed
```

```

File: common/build-style/gemspec.sh (hunk 21 above)
Comment (maxice8) [reply common/build-style/gemspec.sh]: No those are the names ruby picked when being compiled on Void and if you want to change and make them the same them modify the `ruby` package to do so.

Comment (Cogitri) [reply common/build-style/gemspec.sh]: Eh? I'm talking about handling armv6/7hf to the case statement

File: Manual.md (hunk 18 above)
Comment (Cogitri) [reply Manual.md]: Oh, missed the "provides", nevermind my 2nd comment then

File: common/build-style/gem.sh (hunk 20 above)
Comment (Cogitri) [reply common/build-style/gem.sh]: Uh?

Comment (maxice8) [reply common/build-style/gem.sh]: Yeah sorry, the verbose output shows all the files to be installed

File: Manual.md [hunk 23]
```diff
@@ -781,6 +781,17 @@ with the character `r` in the `version` variable. The `distfiles`
 location will automatically be set as well as the package made to depend
 on `R`.
 
+- `gemspec` For packages that use
+[gemspec](https://guides.rubygems.org/specification-reference/) files, this is necessary
```
Comment (Chocimier) [inline Manual.md]: I would replace semicolons with full stops in this sentence. Now for me it looks like:

For \<some condition\>, this is necessary that \<things i need to do\>.

File: Manual.md [hunk 24]
```diff
@@ -781,6 +781,17 @@ with the character `r` in the `version` variable. The `distfiles`
...
+for ruby projects that provide extensions with compiled code (like C), if your ruby project
+does not provide any of those then consider using the `gem` build style instead.
+
+- `gem` For packages that are installed using gems from [RubyGems](https://rubygems.org/)
+or other providers. Packages that use this build style must follow the ruby-`pkgname`
```
Comment (Chocimier) [inline Manual.md]: So *ruby-* should not be part of pkgname? If it should, `ruby-<name>` is clearer and consistent with _naming conventions_ section.

**Comment by Chocimier** [comment]
> reworded, please check again

Now it's ok.

File: Manual.md [hunk 25]
```diff
@@ -781,6 +781,18 @@ with the character `r` in the `version` variable. The `distfiles`
 location will automatically be set as well as the package made to depend
 on `R`.
 
+- `gemspec` For packages that use
+[gemspec](https://guides.rubygems.org/specification-reference/) files for building a ruby
+gem and then installing it. The `gem_cmd` variable can be used to set the name of the gem
+binary, `configure_args` can be used to pass arguments during compilation. If your package
```
Comment (Cogitri) [inline Manual.md]: IMO a full stop is a better fit here than a comma.

File: Manual.md [hunk 26]
```diff
@@ -781,6 +781,18 @@ with the character `r` in the `version` variable. The `distfiles`
...
+does not make use of compiled extensions consider using the `gem` build style instead.
+
+- `gem` For packages that are installed using gems from [RubyGems](https://rubygems.org/).
+The `gem_cmd` variable can be used to set the name of the gem binary. `noarch` is set
+unconditionally and `distfiles` is set by the build style if the template does not so.
+If your gem provides extensions which must be compiled consider using the `gemspec`
+build style instead.
```
Comment (Cogitri) [inline Manual.md]: > If your gem provides extensions 

Still not quite happy with this, please somehow make clear that these are extensions written in languages other than ruby and as such are arch specific.

File: common/build-style/gem.sh [hunk 27]
```diff
@@ -0,0 +1,60 @@
...
+		--ignore-dependencies \
+		--no-document \
+		--verbose \
+		${XBPS_SRCDISTDIR}/${pkgname}-${version}/${pkgname#ruby-}-${version}.gem
+
+	# Remove cache
+	rm -rf ${DESTDIR}/${_GEMDIR}/cache
+
+	# Remove ext directory, they are only source code and configuration
+	# The actual extensions are in a arch path guarded
```
Comment (Cogitri) [inline common/build-style/gem.sh]: an arch path

File: common/build-style/gem.sh [hunk 28]
```diff
@@ -0,0 +1,60 @@
...
+	rm -rf ${_INSTDIR}/ext
+
+	# Remove installed tests and benchmarks
+	rm -rf ${_INSTDIR}/{test,tests,autotest,benchmark,benchmarks,script,examples,demo}
+
+	# Remove files shipped on the root of the gem, most of the time they are useless
```
Comment (Cogitri) [inline common/build-style/gem.sh]: _most_ of the time?

Comment (maxice8) [reply common/build-style/gem.sh]: I have never met one that required it, root is mostly littered with the following:

- git files (.gitignore, .gitmodules)
- README (lol nobody reads that)
- LICENSE/COPYING (We put them in the proper place with vlicense if necessary)
- .gemspec file (Ruby requires them somewhere else)

File: Manual.md (hunk 26 above)
Comment (maxice8) [reply Manual.md]: All extensions by ruby definition are arch specific.

Comment (maxice8) [reply Manual.md]: If it's written in Ruby then it is part of lib/ not ext/

Comment (Cogitri) [reply Manual.md]: Ah, okay.

File: common/build-style/gem.sh (hunk 28 above)
Comment (Cogitri) [reply common/build-style/gem.sh]: Ah, _most_ of the time just sounds as if they're important for some package but we just go ahead and delete them anyway.

**Comment by maxice8** [comment]
Any objections ? Replaced into 4 commits, 2 of them add the build_styles and 2 document them in Manual.md

**Comment by maxice8** [comment]
Tested by packaging loads of stuff

```
srcpkgs/ruby-ansi/
srcpkgs/ruby-contracts/
srcpkgs/ruby-cymbal/
srcpkgs/ruby-facter/
srcpkgs/ruby-git_fame/
srcpkgs/ruby-gpgme/
srcpkgs/ruby-hashie/
srcpkgs/ruby-hiera/
srcpkgs/ruby-hirb/
srcpkgs/ruby-memoist/
srcpkgs/ruby-mercenary/
srcpkgs/ruby-method_profiler/
srcpkgs/ruby-mini_portile2/
srcpkgs/ruby-pkgforge/
srcpkgs/ruby-progressbar/
srcpkgs/ruby-scrub_rb/
srcpkgs/ruby-thor/
srcpkgs/ruby-trollop/

ruby-ansi-1.5.0_1.noarch.xbps
ruby-chronic-0.10.2_1.noarch.xbps
ruby-contracts-0.16.0_1.noarch.xbps
ruby-cymbal-2.0.0_1.noarch.xbps
ruby-facter-2.5.1_1.noarch.xbps
ruby-ffi-1.9.25_1.x86_64.xbps
ruby-git_fame-2.5.2_1.noarch.xbps
ruby-gpgme-2.0.16_1.aarch64.xbps
ruby-gpgme-2.0.16_1.x86_64.xbps
ruby-hashie-3.6.0_1.noarch.xbps
ruby-hiera-3.4.5_1.noarch.xbps
ruby-hirb-0.7.3_1.noarch.xbps
ruby-hpricot-0.8.6_1.aarch64.xbps
ruby-hpricot-0.8.6_1.x86_64.xbps
ruby-memoist-0.16.0_1.noarch.xbps
ruby-mercenary-0.3.6_1.noarch.xbps
ruby-method_profiler-2.0.1_1.noarch.xbps
ruby-mini_portile2-2.3.0_1.noarch.xbps
ruby-mustache-1.1.0_1.noarch.xbps
ruby-pkgforge-0.23.2_1.noarch.xbps
ruby-progressbar-1.10.0_1.noarch.xbps
ruby-rb-inotify-0.9.10_1.noarch.xbps
ruby-rdiscount-2.2.0.1_1.aarch64.xbps
ruby-rdiscount-2.2.0.1_1.x86_64.xbps
ruby-ronn-0.7.3_1.noarch.xbps
ruby-rrunit-0.0.3_1.noarch.xbps
ruby-ruby_dep-1.5.0_1.noarch.xbps
ruby-scrub_rb-1.0.1_1.noarch.xbps
ruby-trollop-2.9.9_1.noarch.xbps
```

That doesn't include packaging all of `jekyll` dependencies except ruby-on-rails which is massive neither `asciidoctor` `

File: Manual.md [hunk 29]
```diff
@@ -781,6 +781,17 @@ with the character `r` in the `version` variable. The `distfiles`
 location will automatically be set as well as the package made to depend
 on `R`.
 
+- `gemspec` For packages that use
+[gemspec](https://guides.rubygems.org/specification-reference/) files for building a ruby
+gem and then installing it. The gem command can be overridden by `gem_cmd`. `configure_args`
+can be used to pass arguments during compilation. If your package does not make use of compiled
```
[the remark this answers is not in the document]
Comment (maxice8) [reply Manual.md]: It is not passed to gem, there is a `--` which stops option parsing, this is passed to stuff like ruby-gpgme which require --use-system-libraries to be passed to it

File: common/build-style/gemspec.sh [hunk 30]
```diff
@@ -0,0 +1,187 @@
...
+	: ${gem_cmd:=gem}
+	: ${gemspec:=${pkgname#ruby-}.gemspec}
+
+	# Fix packages that keep Gem.gemspec as the name instead of the proper
+	# $pkgname.gemspec
+	if [ -f Gem.gemspec ]; then
+		gemspec=Gem.gemspec
+	fi
+
+	if [ -f .gemspec ]; then
```
[the remark this answers is not in the document]
Comment (maxice8) [reply common/build-style/gemspec.sh]: I prefer to keep it that way, imho it is simpler to work out.

File: common/build-style/gemspec.sh [hunk 31]
```diff
@@ -0,0 +1,187 @@
...
+	done
+
+	rm -rf "${_INSTDIR}/man"
+
+	# Place executables in /usr/bin
+	if [ -d "${_INSTDIR}/bin" ]; then
+		for f in "${_INSTDIR}"/bin/*; do
+			vbin "${f}"
+		done
+	fi
```
[the remark this answers is not in the document]
Comment (maxice8) [reply common/build-style/gemspec.sh]: eh ?

**Pull request merged**